### PR TITLE
 refactor: rename `deno` specifiers to `internal`

### PR DESCRIPTION
**Pull request by crowlKats**
Ahead of work of adding deno specifiers to prevent issues & confusion, and for #17648

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Seems fine to me, @dsherret any objections?

**Review by dsherret — APPROVED** [review]
LGTM

**Comment by sigmaSd** [comment]
lsp clients needs to be updated, go_to_def response for example now returns internal: instead of deno:

File: cli/lsp/language_server.rs [hunk 1]
```diff
@@ -3121,7 +3121,7 @@ impl Inner {
       .performance
       .mark("virtual_text_document", Some(&params));
     let specifier = self.url_map.normalize_url(&params.text_document.uri);
-    let contents = if specifier.as_str() == "deno:/status.md" {
+    let contents = if specifier.as_str() == "internal:/status.md" {
```
Comment (dsherret) [inline cli/lsp/language_server.rs:3124]: This breaks vscode_deno, which does a request to this.

Comment (dsherret) [reply cli/lsp/language_server.rs:3124]: https://github.com/denoland/vscode_deno/blob/4b2787824f03879e2cbbff47c9921110ce1f20fb/client/src/commands.ts#L254

**Comment by dsherret** [comment]
@sigmaSd the only thing I could find in some other clients was https://github.com/denoland/deno/pull/17655#discussion_r1097883848 -- do you know of anything else?

**Comment by sigmaSd** [comment]
go to definition as well fails and I assume go to refenrece and type refernece, their answer have use the deno schema
```
[DEBUG][2023-02-06 21:04:41] .../lua/vim/lsp.lua:1381	"LSP[denols]"	"client.request"	1	"textDocument/definition"	{  position = {    character = 57,    line = 3  },  textDocument = {    uri = "file:///home/mrcool/dev/deno/blog/main.tsx"  }}	<function 1>	7
[DEBUG][2023-02-06 21:04:41] .../vim/lsp/rpc.lua:285	"rpc.send"	{  id = 155,  jsonrpc = "2.0",  method = "textDocument/definition",  params = {    position = {      character = 57,      line = 3    },    textDocument = {      uri = "file:///home/mrcool/dev/deno/blog/main.tsx"    }  }}
[DEBUG][2023-02-06 21:04:41] .../vim/lsp/rpc.lua:388	"rpc.receive"	{  id = 155,  jsonrpc = "2.0",  result = { {      targetRange = {        ["end"] = {          character = 0,          line = 906        },        start = {          character = 0,          line = 12        }      },      targetSelectionRange = {        ["end"] = {          character = 0,          line = 906        },        start = {          character = 0,          line = 12        }      },      targetUri = "internal:/https/deno.land/std/testing/asserts.ts"    } }}
```

**Comment by dsherret** [comment]
It's broken in vscode too. cc @crowlKats -- maybe we should revert this for now

**Comment by ayame113** [comment]
atom-ide-deno will also break, but it's no big deal since Atom is no longer in development. :(
https://github.com/ayame113/atom-ide-deno/blob/22e905715dd15a6dbf5cd0c641064306e4371e78/lib/virtual_documet.ts#L14

**Comment by dsherret** [comment]
Thanks. It was reverted in #17673

**Comment by ayame113** [comment]
Glad to hear that!

**Pull request merged**